src/metrics.c

```diff
@@ -213,7 +214,7 @@ static int metrics_set_gauge(metrics *m, char *name, double val, bool delta) {
  * @arg val The sample to add
  * @return 0 on success.
  */
-int metrics_add_sample(metrics *m, metric_type type, char *name, double val) {
+int metrics_add_sample(metrics *m, metric_type type, char *name, double val, double sample_rate) {
```

Comment:
Just to clarify: do you mean update 'metrics_increment_counter' to accept a sample_rate? 

If so, are you ok for me to investigate and make this change in another pr, specifically addressing the counter type?